web/src/dark_theme.ts

```diff
@@ -22,9 +24,17 @@ export function disable(): void {
         const ls = localstorage();
         ls.set("spectator-theme-preference", "light");
         user_settings.color_scheme = settings_config.color_scheme_values.day.code;
+        realm_logo.render();
     }
 }
 
 export function default_preference_checker(): void {
     $(":root").removeClass("dark-theme").addClass("color-scheme-automatic");
+
+    if (page_params.is_spectator) {
+        const ls = localstorage();
+        ls.set("spectator-theme-preference", "automatic");
+        user_settings.color_scheme = settings_config.color_scheme_values.automatic.code;
+        realm_logo.render();
```

Comment:
Let's extract this into a function since it seems like it would make this easier to look at and of course has the benefit of DRY.

Reply:
Extracted this and other such methods for setting spectator theme into `dark_theme.set_theme_for_spectator()`.